extmod/machine_signal: Add Signal print function.

IhorNehrutsa opened the pull request:
Test code is:
```
from machine import Pin, Signal
signal22= Signal(Pin(22, mode=Pin.OUT))
signal22
signal21= Signal(Pin(21, mode=Pin.IN), invert=True)
signal21
```
Ounput is:
```
Signal(Pin(22))
Signal(Pin(21), invert=True)
```
instead of
```
<Signal>
<Signal>
```

IhorNehrutsa commented:
@jimmo 
Could you give your opinion about this PR?

jimmo commented on extmod/machine_signal.c:
```diff
@@ -114,6 +114,21 @@ STATIC mp_obj_t signal_make_new(const mp_obj_type_t *type, size_t n_args, size_t
 
+STATIC void signal_print(const mp_print_t *print, mp_obj_t self_in, mp_print_kind_t kind) {
+    machine_signal_t *self = MP_OBJ_TO_PTR(self_in);
+    const mp_print_fun_t pin_print_function = (mp_print_fun_t)MP_OBJ_TYPE_GET_SLOT_OR_NULL(mp_obj_get_type(self->pin), print);
```
This first bit can just be

```
mp_obj_print_helper(print, self->pin, PRINT_STR);
```

That will save 56 bytes, bringing this change to +80 bytes, which I think is not unreasonable.

IhorNehrutsa replied:
Thanks.
I hope I understood your suggestion correctly.

jimmo replied:
Yep exactly that, thanks!

jimmo reviewed (commented):
Thanks @IhorNehrutsa -- I think this is useful.

Having this is useful for debugging and we already do it for Pin, so Signal should match. If all this printing support is too much code size we could consider making it a build option (e.g. `MICROPY_PY_MACHINE_DETAILED_PRINTING`) similar to how we do terse/detailed error messages.

projectgus commented:
This is an automated heads-up that we've just merged a Pull Request
that removes the STATIC macro from MicroPython's C API.

See https://github.com/micropython/micropython/pull/13763

A search suggests this PR might apply the STATIC macro to some C code. If it
does, then next time you rebase the PR (or merge from master) then you should
please replace all the `STATIC` keywords with `static`.

Although this is an automated message, feel free to @-reply to me directly if
you have any questions about this.

IhorNehrutsa commented:
@dpgeorge 

Ready

dpgeorge commented:
Thanks for rebasing.

But honestly, the increase in code size is quite substantial for a feature like this that's rarely used/requested so far.

IhorNehrutsa committed:
machine_signal: Add Signal print function.

Test code is:
```
from machine import Pin, Signal
signal22= Signal(Pin(22, mode=Pin.OUT))
signal22
signal21= Signal(Pin(21, mode=Pin.IN), invert=True)
signal21
```
Ounput is:
```
Signal(Pin(22))
Signal(Pin(21), invert=True)
```
instead of
```
<Signal>
<Signal>
```

Signed-off-by: IhorNehrutsa <Ihor.Nehrutsa@gmail.com>
